mcc/got/got/got.go

```diff
@@ -109,11 +109,37 @@ func UpsertChangeList(
 	log.Printf(
 		"Adding %s to change list for TTL %d\n",
 		*val.Name,
-		ttl)
+		ttl,
+	)
 	res = append(res, change)
 	return
 }
 
+// DeleteChangeList generates a list of changes for DELETEing the records in
+// names, according to a common type
+func DeleteChangeList(
+	names []string,
+	typ string,
+	list []*route53.ResourceRecordSet,
+) (res []*route53.Change) {
+	var record *route53.ResourceRecordSet
+	for _, name := range names {
+		for _, i := range list {
+			if *i.Name == name &&
+				*i.Type == typ {
+				record = i
+			}
+		}
```

Comment:
From this point on only the last valid record will be used, corresponding to the last `name` given. This should create one change per `name` (if name exists in the DNS server currently).